### PR TITLE
Lifecycle Toggles: Add tooltips and refactor color imports

### PR DESCRIPTION
**Pull request by liyiy**
## Changes
Addresses #4041 

Updates to the lifecycle toggles:
- Add tooltips (feel free to correct any tooltip definitions, I just used google)
- Update chart colors to be more colorblind accessible! (interesting? fact: 1 in 12 men and 1 in 200 women are colorblind)

Screen Shot 2021-04-28 at 6 06 26 PM

## Checklist

- [x] All querysets/queries filter by Organization, by Team, and by User
- [x] Django backend tests
- [x] Jest frontend tests
- [x] Cypress end-to-end tests
- [x] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

**Comment by paolodamico** [comment]
Great stuff @liyiy! Could be a good opportunity for https://github.com/PostHog/posthog/issues/4041 if it makes sense. Also, tagging @corywatilo here to get his input on the colors, they feel a bit too different to the rest of the app, but I'm sure we can find some color-blind-friendly colors that match better.

Also, worth considering UX in terms of colors (i.e. dormant would probably be redish/orange, new probably green, ...)

**Comment by liyiy** [comment]
> Great stuff @liyiy! Could be a good opportunity for #4041 if it makes sense. Also, tagging @corywatilo here to get his input on the colors, they feel a bit too different to the rest of the app, but I'm sure we can find some color-blind-friendly colors that match better.
> 
> Also, worth considering UX in terms of colors (i.e. dormant would probably be redish/orange, new probably green, ...)

Lol yes! That was the issue I meant to tag, tagged my PR to my PR by accident 😆 And I am open to any suggestions for the colors @corywatilo, I used this [site](https://davidmathlogic.com/colorblind/#%23D81B60-%231E88E5-%23FFC107-%23004D40) to cross reference

**Comment by samwinslow** [comment]
Re: accessibility & colors... one way to visually distinguish bars while staying within brand colors would be patterns, e.g. diagonal stripes or grid-square patterns of various sizes/angles. Possibly the user could toggle this setting for their whole account.

**Comment by paolodamico** [comment]
I like the idea of having an accesible mode, and then we can change all colours to have higher contrast ratio too.

**Comment by samwinslow** [comment]
Referencing what I said in this comment: [https://github.com/PostHog/posthog/issues/4156#issuecomment-829248831](https://github.com/PostHog/posthog/issues/4156#issuecomment-829248831) - [this library](https://airbnb.io/visx) looks awesome, and their built-in Patterns component could work for this. Just a thought!

**Comment by liyiy** [comment]
All great points!! Definitely a useful thing to look into and collaborate with design on while working on #4076 and #4156. In the meantime I used https://colorbrewer2.org/#type=qualitative&scheme=Set2&n=3 as recommended by @corywatilo to update the lifecycle chart colors as a start.

Screen Shot 2021-04-29 at 9 57 55 AM

Screen Shot 2021-04-29 at 9 55 06 AM

File: frontend/src/lib/colors.ts [hunk 1]
```diff
@@ -105,16 +105,16 @@ export function getChartColors(backgroundColor: string): string[] {
 
 export function getBarColorFromStatus(status: string): string {
     if (status === 'new') {
```
Comment (samwinslow) [inline frontend/src/lib/colors.ts:107]: Why not use a case-switch and default?

File: frontend/src/scenes/insights/InsightTabs/TrendTab/TrendTab.scss [hunk 2]
```diff
@@ -5,5 +5,22 @@
         display: flex;
         justify-content: space-between;
         padding-bottom: 4px;
+        align-items: center;
+    }
+
+    .new {
+        background-color: #66c2a5;
```
Comment (samwinslow) [inline frontend/src/scenes/insights/InsightTabs/TrendTab/TrendTab.scss:12]: Nit: this is a duplicate declaration of `return cssHSL(161, 43, 58)` in `lib/colors.ts`... these files could get out of sync if we add new states or change colors again. Would be nice to see CSS vars in both places.

**Comment by paolodamico** [comment]
Love the new colors @liyiy! Only the light blue seems a bit too light on the screenshot? But maybe just cause the bars are too small.

File: frontend/src/scenes/insights/InsightTabs/TrendTab/TrendTab.scss (hunk 2 above)
Comment (liyiy) [reply frontend/src/scenes/insights/InsightTabs/TrendTab/TrendTab.scss:12]: Yeah I agree they should both at least be in hex but I'm not sure about turning them into CSS vars in both places, how do I do that without having to declare CSS variables twice in both `global.scss` and `colors.ts`?

**Comment by liyiy** [comment]
> Love the new colors @liyiy! Only the light blue seems a bit too light on the screenshot? But maybe just cause the bars are too small.

I swapped it out with a darker blue 😄 

Screen Shot 2021-04-29 at 10 26 52 AM

Screen Shot 2021-04-29 at 10 26 12 AM

**Comment by corywatilo** [comment]
Not sure if this is the _before_ or _after_ but my only drive-by comment from this screenshot is that the green and blue feel way too similar to be sitting next to each other. (If this was fixed since then, ignore me!)

image

**Comment by paolodamico** [comment]
Visually this look great, but now I'm worried each color is not clear enough from a UX standpoint, i.e. you could easily confused resurrecting and returning, wdyt? Also the red/orange seems to have changed?

By the way, obviously feel free to do what you think is best, just sharing my perspective if you find it helpful.

**Comment by liyiy** [comment]
I think I'll revert the colors back to the original and allow for whoever picks up #4076 to collaborate with design on this 😄, don't want to go too much into a rabbit hole with this PR on the details of the colors, probably a better discussion for #4076  !

Thanks all for the great inputs though!

**Comment by liyiy** [comment]
Back to original! This was a good discussion though 

Screen Shot 2021-04-29 at 10 53 46 AM

**Review by samwinslow — APPROVED** [review]
I like the tooltips + will merge since the colors have been reverted back to their originals.

**Comment by paolodamico** [comment]
Something seems to have gone wrong in the deployment of this PR. Anyways #4174 is a follow up, so we can fix there.

**Pull request merged**